pkg/cmd/util/flag/duration.go

```diff
@@ -0,0 +1,99 @@
+package flag
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+
+	"k8s.io/apimachinery/pkg/util/duration"
+)
+
+// Wrapper around time.Duration with a parser that accepts days, months, years as valid units.
+type Duration struct {
+	time.Duration
+}
+
+// unit map: symbol -> seconds
+var unitMap = map[string]uint64{
+	"ns": uint64(time.Nanosecond),
+	"us": uint64(time.Microsecond),
+	"µs": uint64(time.Microsecond), // U+00B5 = micro symbol
+	"μs": uint64(time.Microsecond), // U+03BC = Greek letter mu
+	"ms": uint64(time.Millisecond),
+	"s":  uint64(time.Second),
+	"m":  uint64(time.Minute),
+	"h":  uint64(time.Hour),
+	"d":  uint64(24 * time.Hour),
+	"w":  uint64(7 * 24 * time.Hour),
+	"mo": uint64(30 * 24 * time.Hour),
+	"yr": uint64(365 * 24 * time.Hour),
```

Comment:
Issue: Using fixed 30-day months and 365-day years creates imprecise durations:
  - Real months: 28-31 days
  - Leap years: 366 days
  - 1 year ≠ 12 months (360 vs 365 days)

 Recommendation: Document this clearly or consider using more conventional approximations:
  - Month: 30 days (current) - reasonable average
  - Year: Consider 365.25 days for better accuracy with leap years

  Suggested comment, perhaps in doc change as well:
  // Approximate durations for extended units:
  // - month (mo): 30 days (not calendar-aware)
  // - year (yr): 365 days (leap years not considered)
  // These are simplified approximations for TTL purposes.